libcloud/common/base.py

```diff
@@ -76,6 +76,39 @@
 RETRY_FAILED_HTTP_REQUESTS = False
 
 
+class LazyObject(object):
+    """An object that doesn't get initialized until accessed."""
+
+    @staticmethod
+    def _proxy(lazy_cls, *lazy_init_args, **lazy_init_kwargs):
+        class Proxy(lazy_cls, object):
+            lazy_obj = None
+
+            def __init__(self):
+                pass
+
+            def __getattribute__(self, attr):
+                lazy_obj = object.__getattribute__(self, 'lazy_obj')
```

Comment:
Changed LazyObject.lazy_obj to LazyObject._lazy_obj.

Regarding the __getattr__ vs __getattribute__ discussion, see http://stackoverflow.com/a/3278104/1436495.
My considerations:
- getattr is only called if attribute resolution fails by normal means
- getattribute is ALWAYS called (even on bound methods); easier to handle given my next point
- the Proxy metaclass inherits from lazy_cls (I want Proxy to be as transparent as possible as a wrapper; I want the returned Proxy instance to have the class info for lazy_cls), meaning that Proxy inherits all methods and class attributes of lazy_cls. Therefore, weird things happen when you call methods intended for the lazy object instead of the Proxy.